pydm/widgets/channel.py

```diff
@@ -46,127 +51,139 @@ class PyDMChannel(object):
+        if self._busy:
+            return
+        self._busy = True
+        data, intro = self.get(with_introspection=True)
+        if not self._use_introspection:
+            intro = self._introspection
+        for mon in self._monitors:
+            try:
+                mon(data=data, introspection=intro)
+            except Exception as ex:
+                logger.exception("Error invoking callback for %r", self)
+        self._busy = False
 
     @property
     def address(self):
         return self._address
 
     @address.setter
     def address(self, address):
-        self._address = clear_channel_address(address)
+        if address is None:
+            conn = self._config.get('connection', None)
+            if conn is not None:
+                self._protocol = conn.get('protocol')
+                self._parameters = conn.get('parameters')
+                self._address = json.dumps(conn)
+            else:
+                self._address = None
+        else:
+            self._address = clear_channel_address(address)
+
+    def connected(self):
+        return self._connected
 
     def connect(self):
         """
         Connect a PyDMChannel to the proper PyDMPlugin
         """
         if not self.address:
             return
-        if is_qt_designer() and not config.DESIGNER_ONLINE:
+        if is_qt_designer() and not PYDM_CONFIG.DESIGNER_ONLINE:
             return
         logger.debug("Connecting %r", self.address)
         # Connect to proper PyDMPlugin
         try:
-            pydm.data_plugins.establish_connection(self)
+            data_plugins.establish_connection(self)
+            self._connected = True
         except Exception:
-            logger.exception("Unable to make proper connection "
-                             "for %r", self)
+            logger.exception("Unable to make proper connection for %r", self)
 
     def disconnect(self, destroying=False):
         """
         Disconnect a PyDMChannel
         """
-        if is_qt_designer() and not config.DESIGNER_ONLINE:
+        if is_qt_designer() and not PYDM_CONFIG.DESIGNER_ONLINE:
             return
         try:
-            plugin = pydm.data_plugins.plugin_for_address(self.address)
+            plugin = data_plugins.plugin_for_address(self.address)
             if not plugin:
                 return
             plugin.remove_connection(self, destroying=destroying)
         except Exception as exc:
-            logger.exception("Unable to remove connection "
-                             "for %r", self)
+            logger.exception("Unable to remove connection for %r", self)
+        self._connected = False
+
+    def get_introspection(self):
+        if self._use_introspection:
+            return DataStore().introspect(self.address)
+        return self._introspection
+
+    def get(self, with_introspection=False):
+        ret = DataStore().fetch(self.address, with_introspection)
+        if not with_introspection:
```

Comment:
Case in point for the return type stability comment above.

Reply:
Addressed at latest commit.